runtime/local/src/lib.rs

```diff
@@ -269,18 +269,25 @@ parameter_types! {
 
 type NegativeImbalance = <Balances as Currency<AccountId>>::NegativeImbalance;
 
-pub struct OnBlockReward;
-impl OnUnbalanced<NegativeImbalance> for OnBlockReward {
-    fn on_nonzero_unbalanced(amount: NegativeImbalance) {
-        let (dapps, maintain) = amount.ration(50, 50);
+pub struct DappsStakingTvlProvider();
+impl Get<Balance> for DappsStakingTvlProvider {
+    fn get() -> Balance {
+        DappsStaking::tvl()
```

Comment:
Is the TVL getter just for the client app, or is it integral to the internal function? Because generally, we use an indexer for aggregating on-chain data for clients, so the chain doesn't have to do too many calculations.

Reply:
No, it's for runtime only.

This is just dependency injection into `block-rewards` pallet since we don't want to make it directly dependent on the `dapps-staking` pallet.